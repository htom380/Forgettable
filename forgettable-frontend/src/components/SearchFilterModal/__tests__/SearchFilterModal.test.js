import TestRenderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import {render, fireEvent, cleanup} from '@testing-library/react';
import {
  within,
  getByText,
  getByTestId,
} from '@testing-library/dom';
import SearchFilterModal from '../SearchFilterModal';

it('renders SearchFilterModal with correct hierarchy', () => {
  const renderer = new ShallowRenderer();

  renderer.render(<SearchFilterModal
    open={true}
    selectedFilter=""
    setSelectedFilter={() => {}}
  />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
