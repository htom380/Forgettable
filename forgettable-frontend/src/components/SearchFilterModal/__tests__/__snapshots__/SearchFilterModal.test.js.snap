// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SearchFilterModal with correct hierarchy 1`] = `
<div
  className="homeFilter 
        null"
>
  <ForwardRef(Box)
    className="searchFilterBox"
  >
    <div>
      <h3>
        Filters
      </h3>
      <div
        className="filterSelectionPanel"
      >
        <CustomButton
          btnText="Encounters"
          className="deselectedButton"
          onClick={[Function]}
        />
        <CustomButton
          btnText="People"
          className="deselectedButton"
          onClick={[Function]}
        />
      </div>
    </div>
  </ForwardRef(Box)>
</div>
`;

exports[`snapshot with no content matches 1`] = `
<div
  className="homeFilter 
        null"
>
  <ForwardRef(Box)
    className="searchFilterBox"
  >
    <div>
      <h3>
        Filters
      </h3>
      <div
        className="filterSelectionPanel"
      >
        <CustomButton
          btnText="Encounters"
          className="deselectedButton"
          onClick={[Function]}
        />
        <CustomButton
          btnText="People"
          className="deselectedButton"
          onClick={[Function]}
        />
      </div>
    </div>
  </ForwardRef(Box)>
</div>
`;
